odoo/http.py

```diff
@@ -928,14 +969,33 @@ def get(self, sid):
                     os.mkdir(dirname, 0o0755)
             with contextlib.suppress(OSError):
                 os.rename(old_path, session_path)
-        return super().get(sid)
-
-    def rotate(self, session, env):
-        self.delete(session)
-        session.sid = self.generate_key()
+        session = super().get(sid)
+        self.delete_old_sessions(session)
+        return session
+
+    def rotate(self, session, env, soft=False):
+        if soft:
+            # Reference old session to new one
+            static = session.sid[:STORED_SESSION_BYTES]
+            previous_sid = session.sid
+            next_sid = static + self.generate_key()[STORED_SESSION_BYTES:]
+            if 'next_sid' in session:
+                next_sid = session['next_sid']
```

Comment:
For what is this ? To cover write concurrency issues ? It might be worth a comment.

In addition, if we fall in this case `if 'next_sid' in session:`, is it worth to do `next_sid = static + self.generate_key()[STORED_SESSION_BYTES:]` just above ? It would be computed for no reason then ?